ember-app/app/initializers/sockets.js

```diff
@@ -6,12 +6,13 @@ export function initialize(container, application){
   let socket = Ember.Object.extend({
     correlationId : correlationId,
     sockets: {},
+    publish: Ember.K,
     subscribe: Ember.K,
     unsubscribe: Ember.K,
     subscribeTo: Ember.K
   });
 
-  if(application.get("socketBackend")){
+  if(application.get("socketBackend") &&  window.Faye){
```

Comment:
Would it be more durable to check if `typeof faye === 'function'`, since that's specifically what we need? Or should the `window.Faye` check live in `faye()`?